spec/models/itt_provider_form_spec.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.describe IttProviderForm, type: :model do
+  describe '#save' do
+    context 'when itt_provider_enrolled is missing' do
+      it 'returns false and has errors' do
+        form = described_class.new
+
+        expect(form.save).to eq false
+        expect(form.errors.full_messages).to eq [
+             'Itt provider enrolled Tell us if you’ve been enrolled in initial teacher training',
+           ]
+      end
```

Comment:
I generally subscribe to the single assertion per test approach. This would be easy to change to align with that.

We can also use the shoulda_matchers to one line these.

```suggestion
      subect(:form) { described_class.new }
      
      specify do
        expect(form)
          .to validate_presence_of(:itt_provider_enrolled)
          .with_message('Itt provider enrolled Tell us if you’ve been enrolled in initial teacher training')
      end
```

Reply:
That code isn't doing the same thing though, is it? It's checking for validations but it's not checking if the save method is running those validations. I was thinking that testing it like that would be longer than doing it "in line" as I did it.

I don't have a dog in the multiple assertions race; happy either way, but generally prefer to rely less on `subject` and `before` hooks and other things as I think it makes some tests harder to figure out than "just read everything that happens in the `it` block". That's personal preference and I'm happy to align one way or another.

Reply:
This is not a blocker for me, so consider the below as me articulating how I think about tests rather than a request for a change. I'm happy to go with your approach if you prefer this after reading my thoughts.

I take your point that these tests aren't the same. However, I think this speaks to a more subtle difference here.

My preference for the single assertion is related to the single responsibility principle that I find to be very useful across all areas of software development.

When a test/method/controller/PR is responsible for one thing, I find it much easier to reason about.

Sure, we can lose the benefit of this simplicity when we overload `subject` and `before` blocks. I think this is more incorrect usage of those tools rather than them being bad.

I think the specifics of the validations are a separate concern from testing that `save` runs validations.

I'd write individual tests for each validation and then a separate describe block for `#save`. With assertions for the return value and also that the `errors` method populates.